libs/sdk-backend-tiger/src/auth.ts

```diff
@@ -72,15 +72,29 @@ export abstract class TigerAuthProviderBase implements IAuthenticationProvider {
  * @public
  */
 export class TigerTokenAuthProvider extends TigerAuthProviderBase {
+    // TigerBackend has factory methods that will return new instance of itself recreated with provided
+    // parameters. AuthProvider instance (this class) is forwarded to the new instance but client is
+    // recreated. Various components keep reference to previous instances of TigerBackend that uses the
+    // previous clients instances to make Axios calls. This could be probably bug in some cases.
+    // These clients would not get token updated and if token has expiration, the next API call would
+    // fail and application would logout the user from the session. The token auth provider must keep the
+    // reference to all these clients to update their tokens if needed.
+    private clients: ITigerClient[] = [];
+
     public constructor(
-        private readonly apiToken: string,
+        private apiToken: string,
         private readonly notAuthenticatedHandler?: NotAuthenticatedHandler,
     ) {
         super();
     }
 
     public initializeClient(client: ITigerClient): void {
         setAxiosAuthorizationToken(client.axios, this.apiToken);
+
+        // Keep reference to each new unique tiger client instance for token re-initialization purposes.
+        if (!this.clients.includes(client)) {
+            this.clients.push(client);
```

Comment:
no clean up is needed during deauthentication?

Reply:
I think you can use the same client to suthenticate again.